### PR TITLE
Replace zones flag with additional filter

### PR DESCRIPTION
**Pull request by azylinski**
Fix WARNING:

Flag `--zones` is deprecated. Use `--filter="zone:( ZONE ... )"` instead.
For example `--filter="zone:( europe-west1-b europe-west1-c )"`.

### CODE REVIEW

**Review by alvaroaleman — COMMENTED** [review]
Deferring to Jeff, I am not really familiar with the janitors
/ok-to-test
/uncc

File: cmd/janitor/gcp_janitor.py [hunk 1]
```diff
@@ -256,14 +256,17 @@ def collect(project, age, resource, filt, clear_all):
     if resource.name == 'sinks' and not clear_all:
         return col
 
+    if resource.condition == 'zone' and resource.name not in ['sole-tenancy', 'network-endpoint-groups']:
+        if len(filt) > 0:
+            filt.append(' AND ')
+        filt.append('zone=(%s)' % ','.join(ZONES))
```
Comment (ixdy) [inline cmd/janitor/gcp_janitor.py:262]: are these supposed to be comma-separated or space-separated?

File: cmd/janitor/gcp_janitor.py [hunk 2]
```diff
@@ -256,14 +256,17 @@ def collect(project, age, resource, filt, clear_all):
     if resource.name == 'sinks' and not clear_all:
         return col
 
+    if resource.condition == 'zone' and resource.name not in ['sole-tenancy', 'network-endpoint-groups']:
+        if len(filt) > 0:
+            filt.append(' AND ')
```
Comment (ixdy) [inline cmd/janitor/gcp_janitor.py:261]: isn't `filt` a string? don't you want `+=` here (and the next line)?

**Review by ixdy — COMMENTED** [review]
Have you tested this change locally? Our automated testing doesn't really test the janitor.

Does this work with all resource types? I'm concerned whether there are any that only support `--zones` and not specifying the `zone` in the filter.

/ok-to-test

**Comment by ixdy** [comment]
/close

I think this was obsoleted by #98.

**Pull request closed without merging**